File: backend/app/models/lesson.py
from pydantic import BaseModel

class LessonRequest(BaseModel):
    subject: str
    topic: str
    grade: int
    duration_minutes: int

class Activity(BaseModel):
    name: str
    duration_minutes: int
    instructions: str

class LessonResponse(BaseModel):
    subject: str
    topic: str
    title: str
    objectives: list[str]
    materials: list[str]
    activities: list[Activity]

class SavedLesson (BaseModel):
    id: int
    subject: str
    topic: str
    grade: int
    duration_minutes: int
    title: str
    objectives: list[str]
    materials: list[str]
    activities: list[Activity]

File: frontend/src/types/lesson.ts
export type Activity = {
  name: string;
  duration_minutes: number;
  instructions: string
};

export type Lesson = {
  id: number;    // this means that id is optional
  subject: string;
  topic: string;
  grade: string;
  duration_minutes: number;
  title: string;
  objectives: string[];
  prior_knowledge: string[];
  materials: string[];
  activities: Activity[];

  reflection: Reflection | null;
};

export type LessonRequest = {
  subject: string;
  topic: string;
  grade: number;
  duration_minutes: number
}

export type ActivityReflection = {
  activity_index: number;
  rating: number | null;
  notes: string;
};

export type Reflection = {
  lesson_id: number;

  objectives_rating: number | null;
  objectives_notes: string;

  prior_knowledge_rating: number | null;
  prior_knowledge_notes: string;

  materials_rating: number | null;
  materials_notes: string;

  activities: ActivityReflection[];
  keep_notes: string;
  change_notes: string;
}